src/hooks/useCountdown.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+
+export type Time = string | number | Date;
+
+export function useCountdown(time: Time) {
+  const [countdown, setCountdown] = React.useState(timeDifference(time));
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => setCountdown(timeDifference(time)), 1000);
```

Comment:
do you think its worth it to pull this timer out into its own component. so lets say if you wanted to add other time based things, like refreshing from api, you would only need to instanciate a single set timeout that other components could hook into?

Reply:
We could extract this into a `useInterval` hook that would be the "hook version" of `setInterval`. Basically this: https://github.com/donavon/use-interval

Reply:
yea thats not a bad idea, though i dont think you need to change this just yet unless you want to